src/test/java/org/folio/services/iteration/IterationServiceTest.java

```diff
@@ -0,0 +1,71 @@
+package org.folio.services.iteration;
+
+import static org.folio.rest.api.TestBase.get;
+import static org.folio.rest.jaxrs.model.IterationJob.JobStatus.IN_PROGRESS;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.UUID;
+import java.util.function.UnaryOperator;
+
+import io.vertx.core.Future;
+import org.folio.persist.IterationJobRepository;
+import org.folio.rest.jaxrs.model.IterationJobParams;
+import org.junit.Before;
+import org.junit.Test;
+
+public class IterationServiceTest {
+
+  private IterationJobRepository repository;
+  private IterationJobRunner runner;
+  private IterationService service;
+
+  @Before
+  public void setUp() {
+    repository = mock(IterationJobRepository.class);
+    runner = mock(IterationJobRunner.class);
+    service = new IterationService(repository, runner);
+  }
+
+  @Test
+  public void canSubmitIteration() {
+    when(repository.save(any(), any()))
+        .thenReturn(Future.succeededFuture(UUID.randomUUID().toString()));
+
+    IterationJobParams jobParams = new IterationJobParams()
+      .withEventType("ITERATE")
+      .withTopicName("inventory.instance.iteration");
+
+    var job = get(service.submitIteration(jobParams));
+
+    assertThat(job.getId(), notNullValue());
+    assertThat(job.getJobStatus(), is(IN_PROGRESS));
+    assertThat(job.getPublished(), is(0));
+    assertThat(job.getSubmittedDate(), notNullValue());
+    assertThat(job.getJobParams(), is(jobParams));
+
+    verify(runner, times(1)).startIteration(any());
```

Comment:
Does it not matter what parameters the iteration job the parameters is started with?